File: 904-leaf-similar-trees/leaf-similar-trees.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean leafSimilar(TreeNode root1, TreeNode root2) {
        ArrayList<Integer> l=new ArrayList<>();
        ArrayList<Integer> m=new ArrayList<>();
        f(root1,l);
        f(root2,m);
        return l.equals(m);
    }
    static void f(TreeNode r1,ArrayList<Integer> l){
        if(r1.right==null && r1.left==null){
            l.add(r1.val);
            return;
        }
        else if (r1.left==null){
            f(r1.right,l);
        }
        else if(r1.right==null){
            f(r1.left,l);
        }
        else{
            f(r1.right,l);
            f(r1.left,l);
        }
    }
}

File: 904-leaf-similar-trees/leaf-similar-trees.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def leafSimilar(self, root1, root2):
        l1=[]
        l2=[]
        self.f(root1,l1)
        self.f(root2,l2)
        return l1==l2
    def f(self,r,l):
        if r==None:
            return
        if r.left==None and r.right==None:
            l.append(r.val)
        self.f(r.right,l)
        self.f(r.left,l)
        